### ISSUE TITLE
Add support for smartchef V2 aka SC02

### ISSUE THREAD

**Issue by JamesRH** [graphefruit/Beanconqueror#807]
The inexpensive smartchef bluetooth scale that was listed on US amazon for $15 recently is no longer the same hardware version as the smartchef supported by this Beanconqueror, despite the image and box looking like the old version (the device inside does not have the chef logo engraved in the metal scale surface). The model numbers have also not changed, although the device is totally different.
https://www.amazon.com/dp/B009LCM90C/

The new version advertises itself as SC02 over BLE, which seems like a good way to refer to it as the manufacturer does not otherwise specify any difference. 

Multiple online forums seem to indicate that the old version has been replaced with this version from most retailers. One with some useful information on the Bluetooth difference is https://www.reddit.com/r/espresso/s/OD8Qzq0ZjG

The final comment in an issue (closed when the old version was supported) indicates that a new issue will be opened
https://github.com/graphefruit/Beanconqueror/issues/566#issuecomment-1660906377
I could not find it and so opened this issue.

This comment on the GitHub issues page of a different espresso scale app that is adding the SC02 to their code base provides some technical details on what seems to be the only differences with the new scale
https://github.com/decentespresso/de1app/pull/272#issuecomment-1691534866
Basically, part of the raw hex data indicates a multiplier that sets the order of magnitude of the encoded number that follows. 

A BLE scanner of the SC02 I have reports the following:
    Device Address : 50:FB:19:87:D1:CC
    Device Name : SC02
    Rssi : -63 dBm
    Raw Data : 14 FF CA 10 0F 03 02 00 00 00 5A 00 05 00 50 FB 19 87 D1 CC 69 05 09 53 43 30 32
It is marked as not connectable. The raw data values change with the weight of the scale.

Some names and numbers associated with the V2 scale, some or all may ay be identical with the V1:
    Smartchef.me
    Smart Chief smart food scale
    Precision series RX402b20
    Designed by ReFleX
    Fcc ID: PS6-RX402b (not a real FCC id)
    Super egg scale
    SKU 701748801781
    UPC 799975126470 701748801781

**Comment by JamesRH** [comment]
I would be happy to try to contribute to address this issue, although my coding experience is mostly in Python, R, and the auduino varient of C/C++

I would also be happy to help test APKs, or even build from a development branch, although I have never used your build framework before. 

Let me know how you would like me to proceed.

**Comment by graphefruit** [comment]
Hello @JamesRH ,
Thanks for your ticket and your support here!
We've had a look at the V2 back then (more then one year ago?).
The issue we've found back then was that the V2 advertises the weight completely different then what is actually implemented in Beanconqueror, and even we would need a new library/plugin that it would work.

Therefore we didn't dig any deeper.

Also I tried to establish contact to Smart Chef more then 6 months without success.

I'd be happy to merge the PR if its working, I can't offer time investment on development side, but I try to support you here as good as possible.

(As a other solution you could just buy the 15$ alibaba scale which is also supported n BQ)


Have a great cup of coffee
Lars

**Comment by JamesRH** [comment]
Which Alibaba scale?

**Comment by graphefruit** [comment]
> Which Alibaba scale?

#603

**Comment by graphefruit** [comment]
Hello @JamesRH,
did you had any progress with the Smartchef V2?

**Comment by graphefruit** [comment]
I gonna close this ticket after its becoming idle.
Please open up a new one if needed.
